mlx/ops.h

```diff
@@ -115,6 +115,13 @@ inline array identity(int n, StreamOrDevice s = {}) {
 /** Reshape an array to the given shape. */
 array reshape(const array& a, std::vector<int> shape, StreamOrDevice s = {});
 
+/** Flatten an array to a 1D array. */
+array flatten(
+    const array& a,
+    int start_dim = 0,
+    int end_dim = -1,
```

Comment:
Following `numpy` we should not take a `start_dim` and `end_dim` in the `flatten` API. That should simplify the implementation a lot (you can get rid of the `getNewShape` stuff and just call `reshape` with a `{-1}`.

Reply:
Updating the flattening procedure through the exclusive use of `reshape(a, {-1} ...)`, consistently results in a 1D array. This differs from the behavior of the `flatten` operation. Consider a scenario involving a batch of images with the shape `[16, 3, 28, 28]`; the objective is to flatten the tensor from the second dimension onward, while preserving the batch dimension.

In a broader context:
The flattening process transforms the input by reshaping it into a one-dimensional tensor. If specific start_dim or end_dim parameters are provided, only dimensions ranging from start_dim to end_dim (inclusive) are flattened.

Reply:
Ok, I think this is sufficiently useful that we can keep it.